### PR TITLE
Cache dependencies on travis

### PR DESCRIPTION
**Pull request by sol**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -12,18 +14,23 @@ addons:
...
+      - ghc-7.10.2
       - cabal-install-1.22
 
 before_install:
   - export PATH=/opt/ghc/$GHCVER/bin:/opt/cabal/1.22/bin:$PATH
+  - (mkdir -p $HOME/.local/bin && cd $HOME/.local/bin && wget https://zalora-public.s3.amazonaws.com/tinc && chmod +x tinc)
   - ghc --version
   - cabal --version
   - travis_retry cabal update
+  - sed -i 's/^jobs:/-- jobs:/' ${HOME}/.cabal/config
```
Comment (soenkehahn) [inline .travis.yml:26]: This disables concurrent builds, doesn't it? Why is that necessary?

Comment (sol) [inline .travis.yml:26]: Without it performance degrade considerably.

**Comment by soenkehahn** [comment]
Thanks!
